api/v3/utils.php

```diff
@@ -2100,6 +2103,50 @@ function _civicrm_api3_validate_integer(&$params, $fieldName, &$fieldInfo, $enti
   }
 }
 
+/**
+ * Helper function to determine country_id given the myriad of values for country_id or country that are supported
+ * @param $params
+ *
+ * @return int|null
+ */
+function _civicrm_api3_resolve_country_id($params) {
+  if (!empty($params['country_id'])) {
```

Comment:
I think we have some cached functions that might be better BUT I'm happy to merge this when it passes tests & iterate for performance

Reply:
Oooh, please teach me the best way to cache a variable for the page request only in CiviCRM :) Do you'll use static variables like drupal_static() in D7 or got something else?